pkg/kubelet/cm/dra/manager.go

```diff
@@ -207,11 +207,11 @@ func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
 			continue
 		}
 
-		// Delete pod UID from the cache
-		claimInfo.deletePodReference(pod.UID)
-
 		// Skip calling NodeUnprepareResource if other pods are still referencing it
-		if len(claimInfo.podUIDs) > 0 {
+		if len(claimInfo.podUIDs) > 1 {
+			// Delete pod UID from the cache if it's not the last pod
+			// referencing the claim
```

Comment:
I think this comment is superfluous, please remove it.

Reply:
done